### PR TITLE
Mimite Runtime Fixes, Mob Balancing, and more.

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

Turns out there where a few missed runtimes, and some issues with venting and travelling across the station.
This fixes those, nerfs the mimites HP and on-click ambush, increases the time between replication, forces a vent on replication so they dont stack up, sets a few backup items to change into if nothing is nearby, and more.

Full list of changes-
Shorter Announce Time
Announce to ghosts
Increased original spawn from 2 to 3
Updated Annoucement message
HP down from 75 to 50
Increased range at which Examine shows something odd
Increased Replication time
Ambush venom amount nerfed
Grow_As is now weighted, so less ambush crates.
Two procs, ApproachVent and TryVentcrawl, to fix venting issues that came with the spiderling code.
Attempt to vent after replication to prevent mob buildup in one location.

Global list for all mimites
Endwhen-60 min
If no mimites exist anymore, announce it
If they exist at the 60 min mark, replication stops completely
Each mimite has a fixed replication cap, each time one replicates, its child inherits the new cap - 1

## Why It's Good For The Game

Fixes good

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

(https://github.com/user-attachments/assets/25ae2b47-f482-49d7-864a-c02e145913bf)

</details>

## Changelog
:cl:
add: If all Mimites are dead, an announcement plays.
add: If the Mimite event hits 60 minutes and mimites still alive, they stop replicating entirely
tweak: tweaked replication rate of the Mimites, and added a replication limit
balance: Nerfed Mimite health and ambush venom. Remember that they can be scanned with Discovery Scanners and examine can reveal them.
fix: fixed a few runtimes regarding Mimites
/:cl:

### CODE REVIEW

File: code/modules/events/mimite_infestation.dm [hunk 1]
```diff
@@ -23,7 +23,7 @@
 			living_mimites = TRUE
 
 	if(living_mimites || fake)
-		priority_announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", ANNOUNCER_ALIENS)
+		priority_announce("Anomlous lifesigns detected aboard [station_name()]. Please report any strange disturbances and Secure any exterior access, including ducting and ventilation.", "Lifesign Anomaly Alert", ANNOUNCER_ALIENS)
```
Comment (PowerfulBacon) [inline code/modules/events/mimite_infestation.dm]: Revert this, keep the alert ambiguous so people don't know what is spawning

**Review by PowerfulBacon — COMMENTED** [review]
Biggest issue is that theres no way to be sure that you killed them all, and no way to hunt them down once they appear

**Comment by Rukofamicom** [comment]
There should be an upper limit to replication, so that after a certain point no more can spawn.

This could be as simple as a time limit from initial spawn that prevents further reproduction of survivors.

**Comment by ro5490** [comment]
> Biggest issue is that theres no way to be sure that you killed them all, and no way to hunt them down once they appear

I can add an alert for when the threat has been cleared if needed, and maybe remove the huds update so that they show as living creatures for Medhuds?

**Comment by ro5490** [comment]
> There should be an upper limit to replication, so that after a certain point no more can spawn.
> 
> This could be as simple as a time limit from initial spawn that prevents further reproduction of survivors.

Lookin into that now

**Comment by ro5490** [comment]
CC now nukes em from orbit after 90 min (I think its 90 min anyway?)
They're limited to 30, if they hit that or more, they stop replicating until theres only 10 left

**Comment by Rukofamicom** [comment]
I'd rather they simply stop replicating permanently when time limit is reached instead of being actively nuked. I do think the time limit should be more like 30 minutes too. 

I like the 30/10 as well to prevent completely over the top infestations, but 30 minutes feels like a good upper time limit for them to be a real menace before the round can move on.

**Comment by ro5490** [comment]
From testing the originals seem to replicate roughly 2-4 times by the 30~min mark
so around 16-26 total by then depending on their growth rolls

I can change the nuking to a set-replicate FALSE on all living mimites if that works, and drop it from 90 to 30 though, or just have them stop replicating entirely once they hit 30~ish total living

**Comment by Rukofamicom** [comment]
The 30 total could still be an indefinite infestation is my worry

**Comment by ro5490** [comment]
Yeah, I get what you mean by the meta, but if the crew havent come across all of them by 90 minutes, they're probably never gonna find the last few lmao
Ive seen people walk past some in hallways over and over, so fast that it doesnt trigger the aggro and the mimite stays put in the middle of a random hallway xD

Tis why I dropped the 90 min nuking in there, the majority of them will probably get found and killed but otherwise its probably gonna be an indefinite without the nuking, one could always end up in the back of a room which nobody goes in, spawn a chest mimite, then that chest never gets found

**Review by Rukofamicom — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/mimite.dm [hunk 2]
```diff
@@ -66,19 +66,26 @@
...
+	var/mimite_stuck = 0	//If mimite_lastmove hasn't changed, this will increment until it reaches mimite_stuck_threshold
+	var/mimite_stuck_threshold = 10 //if this == mimite_stuck, it'll force the mimite to stop moving
+	var/attemptingventcrawl = FALSE
+	var/eventongoing = TRUE
+	var/remaining_replications = 4
 
 /mob/living/simple_animal/hostile/mimite/Initialize()
 	. = ..()
 	AddElement(/datum/element/point_of_interest)
+	GLOB.all_mimites += src
```
Comment (EvilDragonfiend) [inline code/modules/mob/living/simple_animal/hostile/mimite.dm:79]:
```DM
LAZYADD(GLOB.all_mimites, src)
```

File: code/modules/mob/living/simple_animal/hostile/mimite.dm [hunk 3]
```diff
@@ -203,72 +212,99 @@
...
+				mimite_lastmove = null
+				return 1
+		else
+			mimite_lastmove = null
+	else
+		mimite_lastmove = src.loc
+	return 0
 
 /mob/living/simple_animal/hostile/mimite/death(gibbed)
+	GLOB.all_mimites -= src
```
Comment (EvilDragonfiend) [inline code/modules/mob/living/simple_animal/hostile/mimite.dm:307]:
```DM
LAZYREMOVE(GLOB.all_mimites, src)
```
shouldn't be at `death()`. It should be at `Destroy()`

File: code/modules/events/mimite_infestation.dm [hunk 4]
```diff
@@ -43,7 +43,23 @@
...
+				new_mimite.remaining_replications = 5
 		spawncount--
+		announce_to_ghosts(new_mimite)
 		message_admins("[ADMIN_LOOKUPFLW(new_mimite)] has been spawned by an event.")
 		log_game("mimites where spawned by an event.")
 	return SUCCESSFUL_SPAWN
+
+
+/datum/round_event/mimite_infestation/end()
+	if(LAZYLEN(GLOB.all_mimites) >= 1) //If they're still around at the end of 60 min, turn of replication permanently
```
Comment (EvilDragonfiend) [inline code/modules/events/mimite_infestation.dm:61]: `if(length(GLOB.all_mimites))`

File: code/modules/mob/living/simple_animal/hostile/mimite.dm [hunk 5]
```diff
@@ -203,72 +212,99 @@
...
+		entry_vent = null
+		attemptingventcrawl = FALSE
+		travelling_in_vent = 0
+		return
+
+/mob/living/simple_animal/hostile/mimite/proc/isStuck()
+	//Check to see if the mimite is stuck due to things like windows or doors or windowdoors
+	if(mimite_lastmove)
+		if(mimite_lastmove == src.loc)
+			if(mimite_stuck_threshold >= ++mimite_stuck)
```
Comment (EvilDragonfiend) [inline code/modules/mob/living/simple_animal/hostile/mimite.dm:296]: `mimite_stuck_threshold` should be a macro.
```DM
#define MIMITE_STUCK_THRESHOLD 10
```

**Review by EvilDragonfiend — COMMENTED** [review]
eh, it got merged. qdel handling is bad.

**Comment by ro5490** [comment]
> eh, it got merged. qdel handling is bad.

Fixing these and opening a fresh PR

**Pull request merged**